### PR TITLE
fix: Missed single pixel instance in upper left corner by `connected_components()`

### LINKED ISSUE #3169: Single pixel instance in the upper left corner is missed by `kornia.contrib.connected_components`
### Single pixel instance in the upper left corner is missed

The connected component algorithm used starts by placing unique values for every pixel in the foreground (with torch.arange) and zeroing out the background with a mask. The value of the first pixel is 0 (from the arange-function) by default which is confused for background in the spacial case above. The proposed solution is just to add one to all initial values.

### Reproduction steps

```bash
target = torch.tensor(
    [
        [1, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [1, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [1, 0, 1, 0, 1],
    ]
).view(1, 1, 5, 5)
out = connected_components(target)

-->

tensor([[[[ 0,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [10,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [22,  0, 24,  0, 26]]]])
```

### Expected behavior

-->
Maybe:
```
tensor([[[[ 1,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [11,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [23,  0, 25,  0, 27]]]])
```

### Environment

```shell
Not relevant.
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Halyjo**
### Avoid missing a single pixel instance in the upper left corner

Example of issue:
```python
target = torch.tensor(
    [
        [1, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [1, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [1, 0, 1, 0, 1],
    ]
).view(1, 1, 5, 5)
out = connected_components(target)
```
-->

BEFORE:
```bash
tensor([[[[ 0,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [10,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [22,  0, 24,  0, 26]]]])
```

PROPOSED:
-->
```bash
tensor([[[[ 1,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [11,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [21,  0, 23,  0, 25]]]])
```
--- 

The connected component algorithm used starts by placing unique values for every pixel in the foreground (with `torch.arange`) and zeroing out the background with a mask. The value of the first pixel is 0 (from the arange-function) by default which is confused for background in the spacial case above. The proposed solution is just to add one to all initial values.

#### Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)

Fixes #3169 

Small comment here from someone new to `kornia` to the reviewer reading this: I love this project! Thank you all for making, sharing and working on it!

### CODE REVIEW

**Comment by edgarriba** [comment]
you need to update the tests and provide a new test that shows what's this patch is fixing

**Comment by Halyjo** [comment]
I need a bit of help here. The failing tests seem unrelated to the changes.

The changes were made in: 
- `kornia/contrib/connected_components.py` 
- `tests/contrib/test_connected_component.py`


The failing tests happen in `tests/onnx/test_sequential.py` and seem to relate to loading, combining and exporting `onnx` models. 

If anyone understands the connection, please let me know and I'll try to fix it.

**Review by edgarriba — APPROVED** [review]
LGTM. No worries on the other failure cases

**Comment by edgarriba** [comment]
@shijianjian you might want to take a look into the onnx crashes
